doc/man/8/rbd.rst

```diff
@@ -308,7 +308,8 @@ Commands
 :command:`encryption format` *image-spec* *format* *passphrase-file* [--cipher-alg *alg*]
   Formats image to an encrypted format.
   All data previously written to the image will become unreadable.
-  A cloned image cannot be formatted, although encrypted images can be cloned.
+  Image clones are inherently encrypted the same as their parent.
+  A cloned image can only be formatted upon creation. See `clone` command.
```

Comment:
My intention was to do a best effort to prevent the user from doing any IO between cloning and formatting.
You're right that he can do it via the C-API, and the reason for that is that I did not want to create yet a new `rbd_cloneX` API.
For the CLI it is simpler as you can just add a new optional arguments for specifying encryption.

Reply:
If something is not constrained at the lowest level (here it is the librbd API), it is not constrained at all.  It's just a fact of life.  So we should either take the plunge and restrict it for real or back off any best effort stop gaps.

I don't see any strong reason to restrict it for real though.  Quoting myself:

> The user could even populate the clone to some extent before (re-)formatting it, meaning that they would "lose" those blocks on (re-)formatting as the corresponding parent blocks would never be exposed again, but it seems very similar to formatting a flat image to me.

How is I/O between cloning and formatting different from I/O between creating a new (flat) image and formatting?  Yes, it would defeat the purpose of the clone as the parent blocks disappear forever, but it doesn't seem to fundamentally break anything.